src/snowblind/snowblind.py

```diff
@@ -9,92 +9,144 @@
 
 
 class SnowblindStep(Step):
-    spec = """
+    spec = f"""
         min_radius = integer(default=4) # Minimum radius of connected pixels in CR
         growth_factor = float(default=2.0) # scale factor to dilate large CR events
         after_jumps = integer(default=2) # number of groups to flag around saturated cores after a jump
         ring_width = float(default=2.0) # number of pixels to dilate around saturated cores
+        new_jump_flag = integer(default={JUMP_DET}) # DQ flag to set for dilated jumps
     """
 
     class_alias = "snowblind"
 
     def process(self, input_data):
         with datamodels.open(input_data) as jump:
             result = jump.copy()
-            bool_jump = (jump.groupdq & JUMP_DET) == JUMP_DET
-            bool_sat = (jump.groupdq & SATURATED) == SATURATED
+            if hasattr(jump, 'groupdq'):
+                self._has_groups = True
+                bool_jump = (jump.groupdq & JUMP_DET) == JUMP_DET
+                bool_sat = (jump.groupdq & SATURATED) == SATURATED
+            else:
+                self._has_groups = False
+                bool_jump = (jump.dq & JUMP_DET) == JUMP_DET
+                bool_sat = (jump.dq & SATURATED) == SATURATED
```

Comment:
In the case of _rate files currently, only pixels that are saturated in all groups get the SATURATED flag.  This is not good in my view, but what it means in practice is that a snowball saturates some pixels mid-exposure and the _rate file has a slope computed for these pixels due to the first part of the ramp, and the DQ only shows JUMP_DET in the area around the core.  This may make it more difficult to detect snowballs.

Btw, there's an open issue to have this changed.  See spacetelescope/jwst#8124

If the snowball lands in the 1st group, then the core may be saturated (it might not be if NFRAMES is large) for all groups, and so one would get NaNs in the core in the _rate file.  Not sure of the best way to handle this.